### PR TITLE
[healthcheckv2extension] Document that extension is not ready for use

### PR DESCRIPTION
**Pull request by mwear**
**Description:**
Several users have tried to use this extension as the README makes it appear as though it's ready for use. I've updated
the README to include a warning and a note that the README is forward looking. I think it's useful to describe the functionality that the extension will provide when #30673 has been fully decomposed and integrated.

cc: @diranged @twistypigeon

Let me know if stronger wording is needed.

**Link to tracking Issue:** #33301 

**Documentation:** README update

### CODE REVIEW

**Review by crobert-1 — APPROVED** [review]
Since this is a documentation-only change it probably doesn't require a changelog, but it's your call as code owner. 👍

**Review by bogdandrutu — APPROVED** [review]

**Pull request merged**
